service/resources.go

```diff
@@ -13,21 +14,19 @@ import (
 	"github.com/artofimagination/mysql-resources-db-go-service/storage"
 )
 
-func (s *Service) AddResource(ctx context.Context, resource *models.Resource) error {
-	// todo: add resource should be return with the added resource
-	log.Debug(ctx, "Add resource")
+func (s *Service) AddResource(ctx context.Context, resource *models.Resource) (*models.Resource, error) {
 	// Execute function
 	if err := s.mySQLStorage.AddResource(resource); err != nil {
-		return myerrors.WithFields(errors.Wrap(err, "mysql error"), models.HTTPCode, http.StatusInternalServerError)
+		return nil, myerrors.WithFields(errors.Wrap(err, "mysql error"), models.HTTPCode, http.StatusInternalServerError)
 	}
 
-	return nil
+	return resource, nil
```

Comment:
Why did you change this to return the new resource?

Reply:
To be able to check the db really saved the object you would like to save OR if the db creates a uuid for the object during save action you could add it to the resource itself and return with the response.